go/vt/vtgate/evalengine/translate_builtin.go

```diff
@@ -746,3 +748,22 @@ func builtinNullIfRewrite(args []Expr) (Expr, error) {
 	result.Else = args[0]
 	return &result, nil
 }
+
+func builtinIfRewrite(args []Expr) (Expr, error) {
+	if len(args) != 3 {
+		return nil, argError("IF")
+	}
+	var result CaseExpr
```

Comment:
this is a very peculiar way to initialize the struct. Is there a reason why you're not just returning a struct literal?

Reply:
@vmg I copy-pasted the style from `ifnull` and `nullif` but I can clean those up as well.

Reply:
Updated all 3 cases!